### PR TITLE
Add --fusemount (was --fusecmd)

### LINKED ISSUE #2864: plugin/hooks: add appropriate hooks for exposing CVMFS transparently
cc @DrDaveD 

As we discussed during the break at SUG19, we should investigate and understand how to add a set of plugin hooks which can give something like CVMFS the appropriate features to transparently and automatically mount itself into the container. Once we have an understanding of the specific hooks, I will update title and post to reference them

### PR DESCRIPTION
**Pull request by DrDaveD**
**Description of the Pull Request (PR):**

Adds --fusecmd built-in option to the fuse mount plugin capability added in PR #3679. Includes a singularity.conf option to disable --fusecmd, implies the --pid option (so fuse processes that are started automatically get cleaned up on exit), searches for the fuse command in $PATH, and makes sure that the fuse mounts stay in the order in which they were requested.

**This fixes or addresses the following GitHub issues:**

- Fixes #2864

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers, @cclerget, @mem

### CODE REVIEW

**Comment by DrDaveD** [comment]
Closing & reopening to try to get past travis error.

**Comment by DrDaveD** [comment]
@cclerget @GodloveD Could this be marked as a 3.4 milestone?

**Comment by DrDaveD** [comment]
Huh.  Travis worked, but not circleci.  Trying close & reopen.

**Comment by DrDaveD** [comment]
Close & reopen apparently does not retry circleci.  Can somebody please kick it to retry?

File: cmd/internal/cli/action_flags.go [hunk 1]
```diff
@@ -305,6 +306,17 @@ var commonEncryptFlag = cmdline.Flag{
...
 
+// --fusecmd
+var actionFuseCmdFlag = cmdline.Flag{
+	ID:           "actionFuseCmdFlag",
+	Value:        &FuseCmd,
+	DefaultValue: []string{},
+	Name:         "fusecmd",
+	Usage:        "Command to run inside the container to implement a libfuse3-based filesystem. The last parameter is a mountpoint that will be pre-mounted and replaced with a /dev/fd/NN path to the fuse file descriptor.  Implies --pid.",
+	EnvKeys:      []string{"FUSECMD"},
+	ExcludedOS:   []string{cmdline.Darwin},
```
Comment (GodloveD) [inline cmd/internal/cli/action_flags.go:326]:
```suggestion
	ExcludedOS:   []string{cmdline.Darwin},
	Hidden:       true,
```

Comment (DrDaveD) [reply cmd/internal/cli/action_flags.go:326]: Included in my last commit.

**Review by mem — APPROVED** [review]

**Pull request merged**